### PR TITLE
U220/t221 rt meta ball

### PR DESCRIPTION
**Pull request by tofb15**

### CODE REVIEW

File: Sail/src/API/DX12/shader/DX12ConstantBuffer.cpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ namespace ShaderComponent {
 			expanding = true;
 
 			unsigned int oldSize = m_resourceHeapSize;
-			unsigned int size = (nMeshes + 1) * m_byteAlignedSize;
+			unsigned int size = (nMeshes * 2) * m_byteAlignedSize;
```
Comment (Piratkopia13) [inline Sail/src/API/DX12/shader/DX12ConstantBuffer.cpp]: probably fine, but why this?

File: Sail/src/Sail/api/Renderer.h [hunk 2]
```diff
@@ -25,19 +26,37 @@ class Renderer : public IEventListener {
...
 		std::vector<bool> hasUpdatedSinceLastRender;
+
+		union {
+			struct {
+				Mesh* mesh;
+			} model;
+			struct {
+				Material* material;
+			} nonModel;
+		};
```
Comment (Piratkopia13) [inline Sail/src/Sail/api/Renderer.h]: why what who

File: Sail/src/Sail/entities/components/MetaballComponent.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#include "Sail/graphics/geometry/PhongMaterial.h"
+
+class MetaballComponent : public Component <MetaballComponent>{
+
+public:
+	MetaballComponent(float radius = 1.0f);
+	~MetaballComponent();
+	float getRadius() const;
+	void setRadius(float radius);
+	PhongMaterial* getMaterial();
```
Comment (Piratkopia13) [inline Sail/src/Sail/entities/components/MetaballComponent.h]: might need to be replaced with PBRMaterial (?)

Comment (tofb15) [reply Sail/src/Sail/entities/components/MetaballComponent.h]: Not using the metaballs material yet, but i can change it.

**Pull request merged**